apps/src/code-studio/progressRedux.js

```diff
@@ -79,6 +82,21 @@ const initialState = {
   currentPageNumber: PUZZLE_PAGE_NONE
 };
 
+/**
+ * Thunks
+ */
+export function navigateToLevelId(levelId) {
+  return (dispatch, getState) => {
+    const state = getState().progress;
+    const newLevel = state.lessons[0].levels.find(level =>
+      level.ids.find(id => id === levelId)
+    );
+
+    handleNavigateToLevel(state, newLevel.url, levelId);
```

Comment:
minor: maybe this function could be `updateBrowserForLevelNavigation`, or something, just so we're clear that we're manipulating the browser (and any other state management will still happen within redux)

Reply:
Good call.